render/tests/test-event.js

```diff
@@ -45,7 +65,130 @@ o.spec("event", function() {
 		var e = $window.document.createEvent("MouseEvents")
 		e.initEvent("click", true, true)
 		vnode.dom.dispatchEvent(e)
-		
+
+		o(spy.callCount).equals(0)
+	})
+
+	o("removes event when null", function() {
+		var spy = o.spy()
+		var vnode = {tag: "a", attrs: {onclick: spy}}
+		var updated = {tag: "a", attrs: {onclick: null}}
+
+		render(root, [vnode])
+		render(root, [updated])
+
+		var e = $window.document.createEvent("MouseEvents")
+		e.initEvent("click", true, true)
+		vnode.dom.dispatchEvent(e)
+
+		o(spy.callCount).equals(0)
+	})
+
+	o("removes event when undefined", function() {
+		var spy = o.spy()
+		var vnode = {tag: "a", attrs: {onclick: spy}}
+		var updated = {tag: "a", attrs: {onclick: undefined}}
+
+		render(root, [vnode])
+		render(root, [updated])
+
+		var e = $window.document.createEvent("MouseEvents")
+		e.initEvent("click", true, true)
+		vnode.dom.dispatchEvent(e)
+
+		o(spy.callCount).equals(0)
+	})
+
+	o("removes event added via addEventListener when null", function() {
```

Comment:
Some of these tests may be redundant if everything is added with `addEventListener` unless you want to test MouseEvents and TouchEvents.

Reply:
I'd rather stay safe, in case someone down the road decides it's a good idea to start using the raw properties again. That way, we're less likely to see regressions.